### ISSUE TITLE
How can I require an @author tag in every JSDoc?

### ISSUE THREAD

**Issue by riquier** [gajus/eslint-plugin-jsdoc#876]
I am loving this plugin and have been trying to work out if there was a way to require each JSDoc comment to have an @author tag as per our development standards we use?

**Comment by brettz9** [comment]
Glad to hear you're enjoying the plugin.

Yes, you can use the `jsdoc/no-missing-syntax` rule as follows:

```js
`jsdoc/no-missing-syntax`: ['error', {
   contexts: [
     {
       comment: 'JsdocBlock:has(JsdocTag[tag=author])',
       context: 'any'
     }
   ]
 }
]
```

Closing as that should resolve, though feel free to reply further as needed.

**Comment by riquier** [comment]
@brettz9 appreciate the quick reply, I had been trying that though it is only reporting a failure if there are no @author tags in the entire JS file not the `/** **/` block. Any thoughts?

**Comment by brettz9** [comment]
You can use the following which insists that they be at the beginning of the document:

```js
        'jsdoc/require-file-overview': ['error', {
          tags: {
            author: {
              initialCommentsOnly: true,
              mustExist: true,
              preventDuplicates: false,
            },
          },
        }]
```

Although there is no special recognized enforcement for tags ending in `**/`, you could target such tags with:

```
comment: 'JsdocBlock[description=/\\*$/]'
```

**Comment by riquier** [comment]
Apologies explained clumsily, I want to enforce `@author` in every comment block within the file (e.g. every function, method, class), whereas when I tried your solution it was only only checking if `@author` existed in at least 1 comment block in the entire file.

**Comment by brettz9** [comment]
Try `jsdoc/no-restricted-syntax` with:

```js
        {
          contexts: [
            {
              comment: 'JsdocBlock:not(*:has(JsdocTag[tag=author]))',
              context: 'any',
            },
          ],
        },
```

**Comment by riquier** [comment]
that works, really appreciate your help @brettz9
